dimitriosv/week2/index.html

```diff
@@ -0,0 +1,316 @@
+                                        </div>)
+                                }
+                            })}
+                        </div>
+                        <div>Archived:</div>
+                        <div>
+                            {this.props.messages.map(function(message,i) {
+                                if (message.IsArchived) {
+                                    return (<div className="message" key={i}  >
+                                            <Message  Mid={message.messageId} 
+                                                Mtext={message.text}
+                                                MisArchived={message.IsArchived} 
+                                                deleteMessageParent={deleteMessageParent}
+                                                archiveMessageParent={archiveMessageParent}
+                                                addMessageParent={addMessageParent}
+                                                allLists={allLists}
+                                            />
+                                        </div>)
+                                } else {
+                                    ;
+                                }
+                            })}
+                        </div>
+                    </div>
+        },
+         
+
+    });
+
+    var Message = React.createClass({
+        eachList: function(list, i) {  
+        return (
+                 <option key={i} value={list.listId}>{list.listName}</option>
+            );
+        },
+        render: function () {
+            return (<div>
+                        {this.props.MisArchived
+                            ? <div className="archived">
+                            <div className="messagetext" >{this.props.Mtext}</div>
+                            <button type="button" onClick={this.archiveMessageChild.bind(null,this.props.Mid)}>Unarchive</button>
+                            </div>
+                            : <div>
+                            <div className="messagetext" >{this.props.Mtext}</div>
+                            <button type="button"  onClick={this.deleteMessageChild.bind(null,this.props.Mid)}>Delete</button>
+                                <select className="llistcont" defaultValue="0"  onChange={this.moveMessage}>
+                                    <option key="0" value="0" disabled >Move to:</option>
+                                    {this.props.allLists.map(this.eachList)}
+                                 </select>
+                            <button type="button" onClick={this.archiveMessageChild.bind(null,this.props.Mid)}>Archive</button></div>
+                        }
+                   </div>)
+        },
+        moveMessage: function(e) {
+            Listid=e.target.value;  
+            //i am not sure if this is the correct way to communicate with the "grand-parent"
+            this.props.addMessageParent(Listid,this.props.Mtext)        
```

Comment:
I would assume `M` stands for _message_. Please do not use such abbreviations for naming. It makes the code so much harder to read. `messageText` would be the appropriate name here.